src/joint_state_listener.cpp

```diff
@@ -171,8 +171,12 @@ int main(int argc, char** argv)
 
   // gets the location of the robot description on the parameter server
   urdf::Model model;
-  if (!model.initParam("robot_description"))
+  if (!model.initParam("robot_description")) {
+    if (model.links_.empty()) {
+      return 0;
```

Comment:
I'm thinking that we don't want this to be successful; it seems to me that launching robot_state_publisher with a URDF that has no links should probably be louder (use `ROS_ERROR` to print a message), and return an error code (1 seems appropriate here).

Reply:
As mentioned in #81 ROS_ERROR is [already printed by urdfdom](https://github.com/ros/urdfdom/blob/master/urdf_parser/src/model.cpp#L179)